flink-end-to-end-tests/flink-datastream-allround-test/src/main/java/org/apache/flink/streaming/tests/SemanticsCheckMapper.java

```diff
@@ -21,17 +21,15 @@
 import org.apache.flink.api.common.functions.RichFlatMapFunction;
 import org.apache.flink.api.common.state.ValueState;
 import org.apache.flink.api.common.state.ValueStateDescriptor;
-import org.apache.flink.runtime.state.FunctionInitializationContext;
-import org.apache.flink.runtime.state.FunctionSnapshotContext;
-import org.apache.flink.streaming.api.checkpoint.CheckpointedFunction;
+import org.apache.flink.configuration.Configuration;
 import org.apache.flink.util.Collector;
 
 import java.io.Serializable;
 
 /**
  * This mapper validates exactly-once and at-least-once semantics in connection with {@link SequenceGeneratorSource}.
  */
-public class SemanticsCheckMapper extends RichFlatMapFunction<Event, String> implements CheckpointedFunction {
+public class SemanticsCheckMapper extends RichFlatMapFunction<Event, String> {
```

Comment:
Actually the changes here in this file are not strictly required for exactly-once.
I only changed this as part of trying some things out.
Can revert this if you prefer, @StefanRRichter.